### DISCUSSION TITLE
Add pagingation when show the list files

### DISCUSSION THREAD

**Discussion by wooriele** [filebrowser/filebrowser#1609] [Ideas]
Hi,

I think we should add pagingation in the list view, it will be help to improve the performation when we view a huge of files.

Thanks

**Comment by ramiresviana** [comment]
This feature is currently present on the frontend, the number of displayed files increases when scrolling, but additional optimization can be done on the backend.

**Reply by wooriele** [reply]
could you please guide me how to add the optimization on the backend?

**Reply by ramiresviana** [reply]
That will require changes on the source code, the backend is written in go and VueJS is used for the frontend.

**Reply by goni05** [reply]
I think this can be optimized in several ways.  Navigating around the folder structure to list files is one thing, but when moving/copying files, limiting results to folders only would improve the experience.  I really don't need to wait for an entire directory listing to decide where I want to move/copy something.  For file listings, that's likely a bit more complicated, but I can think of several things to "stream" results as they occur, or even try to cache the listings so repeated access will be quicker the next time.  As it is today, even repeated navigation takes a long time, at least on move/copy commands.
